### ISSUE TITLE
execd output with JSON format & use_batch_format = true is broken due to missing trailing newline

### ISSUE THREAD

**Issue by carlosabalde** [influxdata/telegraf#14142]
### Relevant telegraf.conf

```toml
[[inputs.cpu]]
 interval = '1s'

[[outputs.execd]]
 command = ['/usr/bin/bash', '/tmp/foo.sh']
 data_format = 'json'
 use_batch_format = true
```

### Logs from Telegraf

```text
-
```

### System info

Telegraf 1.28.2 + Ubuntu 22.04

### Steps to reproduce

Use the suggested minimal configuration, being `/tmp/foo.sh` something like this:

```
while read line; do
  echo "$line" >> /tmp/foo.log
done < /dev/stdin
```

### Expected behavior

Output logged to `/tmp/foo.log`, one JSON object per line.

### Actual behavior

Nothing logged to `/tmp/foo.log`.

### Additional info

Problem is, [unlike the simple JSON serializer](https://github.com/influxdata/telegraf/blob/release-1.28/plugins/serializers/json/json.go#L75), the [batch JSON serializer](https://github.com/influxdata/telegraf/blob/release-1.28/plugins/serializers/json/json.go#L80) doesn't append a trailing newline. As a result, [the execd output](https://github.com/influxdata/telegraf/blob/release-1.28/plugins/outputs/execd/execd.go#L88) writes the serialized string to the `stdin` of the script, but without a newline, the script will never consume it.

In other words, metrics are pushed to `stdin` as JSON objects, but without newlines is imposible for scripts to know how to delimit those objects. Not sure if the right solution is to add the newline in the batch JSON serializer (looks dirty, but that's current behavior for the simple serializer), or adding it in the execd output.

**Comment by srebhan** [comment]
@carlosabalde can you please test the binary in PR #14150 available once CI finished all tests!? Let me know if this fixes the issue!
